### PR TITLE
More text on ACK rate in the applicability draft

### LINKED ISSUE #330: ACK generation
There is text in the transport document on reducing ACK frequency.  I think that that is sufficient.  As this is more about performance tuning for implementers of the QUIC transport itself, it might be best to remove this section.

### PR DESCRIPTION
**Pull request by mirjak**
fixes #330

### CODE REVIEW

File: draft-ietf-quic-applicability.md [hunk 1]
```diff
@@ -565,11 +565,21 @@ RESET_STREAM frames, which contain only an error code.
 
 # ACK-only packets on constrained links
 
+The stragegy on sending acknowlegdements in QUIC version 1 is adopted from
+TCP, meaning that usually an QUIC implementation is expected to send one
+packet containing an ACK frame for each two packets received.
```
Comment (mirjak) [inline draft-ietf-quic-applicability.md:570]:
```suggestion
packet containing an ACK frame for each two packets received.
```

File: draft-ietf-quic-applicability.md [hunk 2]
```diff
@@ -565,11 +565,21 @@ RESET_STREAM frames, which contain only an error code.
...
+these resources are constrained. With TCP network based mechanisim exists
+that perform so called ACK thining in such situations. As "only-ACK" packets
+are not reliably detecable in QUIC and drop of those packets can have larger
+performance impcats than with TCP, network based ACK thining is not suitable
+for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
+appropriately and some implenentation may even provide an interface to
+impact the amount of ACK traffic sent. Reducing the rate at which
+acknowledgments are generated can preserve constraint resources and
+in such a situation improve overall performance, for both network processing
+as well as application-relevant metrics.
```
Comment (ianswett) [inline draft-ietf-quic-applicability.md:582]:
```suggestion
impact the amount of ACK traffic sent. Reducing the rate at which
acknowledgments are generated in constrained situations can preserve resources
and improve overall performance, for both network processing as well as
application-relevant metrics.
```

File: draft-ietf-quic-applicability.md [hunk 3]
```diff
@@ -565,11 +565,21 @@ RESET_STREAM frames, which contain only an error code.
...
 The cost of sending acknowledgments - in processing cost or link
 utilization - could be a significant proportion of available resources if
-these resources are constrained. Reducing the rate at which acknowledgments
-are generated can preserve these resources and improve overall performance,
-for both network processing as well as application-relevant metrics.
+these resources are constrained. With TCP network based mechanisim exists
+that perform so called ACK thining in such situations. As "only-ACK" packets
+are not reliably detecable in QUIC and drop of those packets can have larger
+performance impcats than with TCP, network based ACK thining is not suitable
+for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
```
Comment (ianswett) [inline draft-ietf-quic-applicability.md:577]:
```suggestion
these resources are constrained.  "ACK-only" packets are not reliably
detectable in QUIC, so network ACK thinning done for TCP is not suitable
for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
```

File: draft-ietf-quic-applicability.md [hunk 4]
```diff
@@ -570,16 +570,14 @@ TCP, meaning that usually an QUIC implementation is expected to send one
...
 The cost of sending acknowledgments - in processing cost or link
 utilization - could be a significant proportion of available resources if
-these resources are constrained. With TCP network based mechanisim exists
-that perform so called ACK thining in such situations. As "only-ACK" packets
-are not reliably detecable in QUIC and drop of those packets can have larger
-performance impcats than with TCP, network based ACK thining is not suitable
+these resources are constrained.  "ACK-only" packets are not reliably
+detectable in QUIC, so network ACK thinning done for TCP is not suitable
 for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
 appropriately and some implenentation may even provide an interface to
```
Comment (ianswett) [inline draft-ietf-quic-applicability.md]:
```suggestion
for QUIC traffic. Instead QUIC endpoints need to control the ACK rate
appropriately and some implenentations may even provide an interface to
```

**Review by ianswett — APPROVED** [review]

File: draft-ietf-quic-applicability.md [hunk 5]
```diff
@@ -565,11 +565,19 @@ RESET_STREAM frames, which contain only an error code.
...
-are generated can preserve these resources and improve overall performance,
-for both network processing as well as application-relevant metrics.
+these resources are constrained.  "ACK-only" packets are not reliably
+detectable in QUIC, so network ACK thinning done for TCP is not suitable
+for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
+appropriately and some implenentation may even provide an interface to
+impact the amount of ACK traffic sent. Reducing the rate at which
+acknowledgments are generated in constrained situations can preserve resources
+and improve overall performance, for both network processing as well as
+application-relevant metrics.
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:580]: I can't suggest over the diff in the way I'd like, but here is my take on this:

```md
# Acknowledgment Efficiency

QUIC version 1 without extensions uses an acknowledgment strategy based on TCP.
That is, every other packet is acknowledged.  However, generating and processing
QUIC acknowledgments can consume significant resources, both in terms of link
utilization and processing time.  Some applications might be able to improve
overall performance by using alternative strategies that reduce the rate of
acknowledgments.
```

This drops the "thinning" text (that's for the other doc).

There is a lot more to unpack here, but the transport doc already talks about this topic in more precise terms.

**Review by martinthomson — COMMENTED** [review]
This seems like it is in the right direction.  I don't like how much text it adds though.

File: draft-ietf-quic-applicability.md [hunk 6]
```diff
@@ -565,11 +565,19 @@ RESET_STREAM frames, which contain only an error code.
...
+TCP, meaning that usually an QUIC implementation is expected to send one
+packet containing an ACK frame for each two packets received.
 The cost of sending acknowledgments - in processing cost or link
 utilization - could be a significant proportion of available resources if
-these resources are constrained. Reducing the rate at which acknowledgments
-are generated can preserve these resources and improve overall performance,
-for both network processing as well as application-relevant metrics.
+these resources are constrained.  "ACK-only" packets are not reliably
+detectable in QUIC, so network ACK thinning done for TCP is not suitable
+for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
```
Comment (gorryfair) [inline draft-ietf-quic-applicability.md:575]: /instead QUIC endpoint need/instead a QUIC endpoint needs/

File: draft-ietf-quic-applicability.md [hunk 7]
```diff
@@ -565,11 +565,19 @@ RESET_STREAM frames, which contain only an error code.
...
+appropriately and some implenentation may even provide an interface to
```
Comment (gorryfair) [inline draft-ietf-quic-applicability.md:576]: /some implementation may even provide an interface to impact/implementations provide an interface that can reduce/ ??? ... although I like MT's comment also which avoids this.

File: draft-ietf-quic-applicability.md (hunk 5 above)
Comment (gorryfair) [reply draft-ietf-quic-applicability.md:580]: That looks good, OK also to me.

**Review by gorryfair — COMMENTED** [review]
For me, I think we are now so close ... I'd kind of like to see two paras, since these are two concepts, and it would be great to see both paras in the same section.

**Pull request merged**